packages/browser-wallet/src/shared/utils/log-helpers.ts

```diff
@@ -0,0 +1,71 @@
+import { storedLog } from '@shared/storage/access';
+import { addToListMaxSize } from '@shared/storage/update';
+import { isDevelopmentBuild } from './environment-helpers';
+
+const loggingLock = 'concordium_log_lock';
+
+// Determines the maximum number of log entries that is stored
+// at a time.
+const LOG_MAX_ENTRIES = 100;
+
+export enum LoggingLevel {
+    INFO = 'INFO',
+    WARN = 'WARN',
+    ERROR = 'ERROR',
+}
+
+export interface Log {
+    timestamp: number;
+    level: LoggingLevel;
+    message: string;
+}
+
+function isError(error: unknown): error is { message: string } {
+    return typeof error === 'object' && error !== null && 'message' in error;
+}
+
+function logForDevelopmentBuild(logEntry: Log) {
+    const logMessage = `[${new Date(logEntry.timestamp).toISOString()}] ${logEntry.level} ${logEntry.message}`;
+    switch (logEntry.level) {
+        case LoggingLevel.WARN:
+            // eslint-disable-next-line no-console
+            console.warn(logMessage);
+            break;
+        case LoggingLevel.ERROR:
+            // eslint-disable-next-line no-console
+            console.error(logMessage);
```

Comment:
Could be nice to have stack trace here if its an error.

Reply:
Let's do that soonTM